src/govuk/components/character-count/template.njk

```diff
@@ -3,10 +3,14 @@
 {% from "../textarea/macro.njk" import govukTextarea %}
 {% from "../hint/macro.njk" import govukHint %}
 
+{%- set hasNoLimit = (not params.maxwords and not params.maxlength) %}
+{%- set fallbackHintText = params.fallbackHintText or 'You can enter up to %{count} ' + ('words' if params.maxwords else 'characters') %}
```

Comment:
If neither maxwords nor maxlength are passed this will always end up talking about characters, but `maxwords` could then be used when the component is initialised in JS.

You can see this if you go to https://govuk-frontend-pr-2915.herokuapp.com/components/character-count/no-maximum/preview and then run `new GOVUKFrontend.CharacterCount($('[data-module]'), { maxwords: 10 }).init()` in the console. The visible message will correctly display the number of words remaining, but if you inspect the fallback hint will still be talking about characters.

Unfortunately I'm not sure that's really fixable with the current approach without making things quite a lot messier… 🤔

Reply:
Yeah, we can only do so much without knowing if it's characters or words that will be counted. In the first take I had set a very generic "You can enter a limited amount of content". Discussing with Beeps and Claire, it doesn't bring anything to the table so left it out.

I may have a tidy way to sort that out: leave the attribute out if no fallback text is set and no maxwords/maxlength are (but keep injecting the default as we do in the HTML from Nunjucks if we have a `maxlength` or `maxwords`). With a default of `''` in the JS, that'd mean no description for the user when they first focus the field, then an announcement as they reach the threshold and the count message starts displaying. This sounds like the tidiest scenario for users.